htmresearch/algorithms/hierarchical_clustering.py

```diff
@@ -0,0 +1,113 @@
+import numpy
+import scipy.sparse
+import scipy.cluster.hierarchy
+import matplotlib.pyplot as plt
+from nupic.algorithms.KNNClassifier import KNNClassifier
+
+
+
+class HierarchicalClustering(object):
+  """
+  Implements hierarchical agglomerative clustering on the output of a classification network.
+  """
+
+
+  def __init__(self, classificationNetwork, linkageMethod='single'):
```

Comment:
Perhaps change the parameter to be an instance of type knn. The experiment runner can extract the KNN from the classificationNetwork and pass it in.  That would make this class a bit more general.